posts/hey-this-looks-familiar-front-end-development-for-back-end-engineers.md

```diff
@@ -0,0 +1,571 @@
+
+![Page and console output with the timeout - messages are printed in the console, but the location name is not changed.](../images/2022/02/fed4bees2-no-data-binding.png)
+
+In order to update the view, we will need to explicitly update the innerHTML whenever we change the data in the state, as we do in the following snippet. We continue to use `that` as an alias for `this`.
+
+``` js
+constructor() {
+  var that = super();
+  // save divElement as part of our object 
+  // so we can manipulate it outside of this function
+  that.divElement = document.createElement('div');
+  // ...
+}
+
+connectedCallback() {
+  var that = this;
+  setTimeout(() => {
+    that.model.location = "Somewhere Else, USA";
+    console.log("Updated location to Somewhere Else!");
+    that.divElement.innerHTML = that.getInnerHTML();
+    console.log("Explicitly set innerHTML of the div element.");
+    console.log("Did the location name change on the webpage?");
+  }, 3000);
+}
+```
+
+With this change made, we see that the location does update to "Somewhere Else, USA" in the output view. 
+
+![Page and console output with the additional timeout - messages are printed in the console, and the location name is changed.](../images/2022/02/fed4bees2-explicitly-update.png)
+
+As it turns out, we don't need this component to update its own data right now. So while there are ways to implement data binding within a component, we won't consider them here. Instead, we'll move on to making our weather card components customizable, so they are able to represent the weather at Ippon's other offices.
+
+## Attribute Binding
+
+We know that we will want to pass the information to populate our model to the component from outside. Let's take a look at what that might look like. The information has to come from somewhere, and right now our only real option is from our `index.html` file. To refresh our memory, here's the `div` we currently have representing the weather card for Washington, DC.
+
+``` html
+<div class="weather-card purple">
+  <h3>In Washington, District of Columbia,<br/>it is currently 30&deg;F and partly cloudy.</h3>
+  <p>Last updated: 2022-01-31 20:30</p>
+</div>
+```
+
+We know we'll need to use the tags for our custom element, `<weather-card-component></weather-card-component>`, somehow incorporating the additional information into or around the tag. The title of this section gives away that we'll eventually bind data to attributes of the element. (Recall that `class` on the `div` element in our HTML snippet is an example of an attribute.) Before we get to the details of attribute binding, I want to briefly present an alternative we won't use in this example and explain why I chose not to use it in this case.
+
+### Templates and Slots
+
+<!-- todo: get more feedback on whether this is a useless diversion (: -->
```

Comment:
Davis left a comment on the preliminary review that this was maybe too much detail for a side-quest. Thoughts from other reviewers?

Reply:
Hmm maybe, but I don't think it was that long of a read. I guess this one is the longest of the posts, so it may be good to cut it.

Reply:
Roger that. I did add a bunch in my soapbox about refactoring and I think that's marginally more useful (lol).

Reply:
I'd made this change in the wrong repo 🙈 but now it's in the right place.